x/observer/keeper/observer_set.go

```diff
@@ -36,23 +37,26 @@ func (k Keeper) IsAddressPartOfObserverSet(ctx sdk.Context, address string) bool
 	return false
 }
 
-func (k Keeper) AddObserverToSet(ctx sdk.Context, address string) {
+// AddObserverToSet adds an observer to the observer set.It makes sure the updated observer set is valid.
+func (k Keeper) AddObserverToSet(ctx sdk.Context, address string) error {
 	observerSet, found := k.GetObserverSet(ctx)
-	if !found {
-		k.SetObserverSet(ctx, types.ObserverSet{
+	switch {
```

Comment:
Why using a switch for a single condition? if is more adapted for this case

Reply:
I would need to use a `if` with an `else` as I think it is a bit cleaner to append only when the set is found.

However, we should call Validate() either way. IMO it looks slightly more readable using the switch

Reply:
```go
func (k Keeper) AddObserverToSet(ctx sdk.Context, address string) error {
	observerSet, found := k.GetObserverSet(ctx)
	if !found {
		observerSet = types.ObserverSet{
			ObserverList: []string{},
		}
	}

	observerSet.ObserverList = append(observerSet.ObserverList, address)
	
	if err := observerSet.Validate(); err != nil {
		return err
	}

	k.SetObserverSet(ctx, observerSet)
	
	return nil
}
```